extensions/bouncer.md

```diff
@@ -0,0 +1,236 @@
+
+From a single bouncer connection, a complete implementation will allow users to:
+* Add, remove, modify and search networks under their bouncer account.
+* List and search channels and queries.
+
+
+## Notes
+References in this document:
+- A `network` is an IRC network that a bouncer connects to
+- A `buffer` is either a channel or query
+- A client is a users client or device that connects to the bouncer
+
+A network is identified by a per-user unique ID that MUST not change. A human readable label MUST also be associated with the network that the user can identify it with in a client UI - this is the `network` tag in most commands. The ID and label may be the same as long as the ID does not change. Neither the ID or label may contain a space or `:` characters.
+
+All commands and data for this bouncer feature MUST be associated to a logged-in and authorised user. It is the implementors choice on how the user authentication is handled and is outside the scope of this document.
+
+## Capabilities
+
+This adds a capability called `BOUNCER`. Once negotiated the server may change it's behaviour to better suit bouncer clients, such as not automatically sending `JOIN` commands for already joined channels.
+
+## RPL_ISUPPORT
+This adds an isupport token called `BOUNCER`. Its optional value is message-tag encoded data including:
+* `network=` The name or label of the network being connected to.
+* `netid=` The ID of the network being connected to.
+
+Eg: `BOUNCER=network=freenode;netid=1234`
+
+## `BOUNCER` command
+This introduces the `BOUNCER` command, followed by a case insensitive subcommand.
+
+All tags sent and received by this command are message-tags encoded.
+
+***Note***: Where a network ID is used to specify a network in a command, a `*` character may be used instead to signify the active network the client is connected to.
+
+
+### Connecting a network
+Syntax: `[c] BOUNCER connect <netid>`
+
+Replies:
+* Various errors:
+  - `[s] BOUNCER connect * :Invalid arguments given`
+  - `[s] BOUNCER connect netid ERR_NETNOTFOUND`
+
+The connection state of the network will be received as mentioned in the network state section.
+
+### Disconnecting a network
+Syntax: `[c] BOUNCER disconnect <netid> [:quit message]`
+
+Replies:
+* Various errors:
+  - `[s] BOUNCER disconnect * ERR_INVALIDARGS`
+  - `[s] BOUNCER disconnect netid ERR_NETNOTFOUND`
+
+The connection state of the network will be received as mentioned in the network state section.
+
+
+### Listing networks
+Syntax: `[c] BOUNCER listnetworks [*]`
+
+This command MUST return all networks under the users bouncer account. The optional second parameter of `*` is a wildcard search string that filters the returned networks. It is matched against the `network` tag of each network.
```

Comment:
I don't think the wildcard search string adds a lot of value here. What's the use-case? Do you expect the user to have a huge number of networks so that filtering can't be done client-side?